### PR TITLE
OPENEUROPA-2473: Add Social media follow paragraph to Content row paragraph 

### PR DESCRIPTION
**Pull request by 22Alexandra**
### Description

Add Social media follow paragraph to Content row paragraph and set Variant field required.

### Change log

- Added: Social media follow paragraph to Content row paragraph
- Changed: Set Variant field required

### Commands

```sh
drush updb

```

### CODE REVIEW

File: oe_paragraphs.post_update.php [hunk 1]
```diff
@@ -74,3 +74,38 @@ function oe_paragraphs_post_update_10002(array &$sandbox): void {
...
+  ];
+  // Add Social media follow paragraph to content row.
+  $field = FieldConfig::load('paragraph.oe_content_row.field_oe_paragraphs');
+  $handler_settings = $field->getSetting('handler_settings');
+  if (isset($handler_settings['target_bundles'])) {
+    $handler_settings['target_bundles']['oe_social_media_follow'] = 'oe_social_media_follow';
+  }
+  // Reorder paragraphs.
+  foreach ($paragraph_weights as $paragraph => $weight) {
+    if (isset($handler_settings['target_bundles_drag_drop'])) {
```
Comment (brummbar) [inline oe_paragraphs.post_update.php]: Let's move this line outside the foreach.

File: oe_paragraphs.post_update.php [hunk 2]
```diff
@@ -74,3 +74,38 @@ function oe_paragraphs_post_update_10002(array &$sandbox): void {
     $entity->save();
   }
 }
+
+/**
+ * Add Social media follow paragraph to Content row paragraph.
```
Comment (brummbar) [inline oe_paragraphs.post_update.php]: Let's merge the two sentences somehow, if possible.

File: oe_paragraphs.post_update.php [hunk 3]
```diff
@@ -74,3 +74,36 @@ function oe_paragraphs_post_update_10002(array &$sandbox): void {
...
+    $handler_settings['target_bundles']['oe_social_media_follow'] = 'oe_social_media_follow';
+  }
+  // Reorder paragraphs.
+  if (isset($handler_settings['target_bundles_drag_drop'])) {
+    foreach ($paragraph_weights as $paragraph => $weight) {
+      $handler_settings['target_bundles_drag_drop'][$paragraph]['weight'] = $weight;
+    }
+  }
+  $field->setSetting('handler_settings', $handler_settings);
+  $field->save();
```
Comment (ademarco) [inline oe_paragraphs.post_update.php]: Let's not add this to the content row, let's only add it to the readme that a new paragraph type is available and, if needed, it needs to be added manually to the content row paragraph.

File: oe_paragraphs.post_update.php [hunk 4]
```diff
@@ -74,3 +74,36 @@ function oe_paragraphs_post_update_10002(array &$sandbox): void {
...
+
+  // Set Variant field required.
+  $field = FieldConfig::load('paragraph.oe_social_media_follow.field_oe_social_media_variant');
+  $field->setRequired(TRUE);
+  $field->save();
```
Comment (ademarco) [inline oe_paragraphs.post_update.php]: This is ok and can stay.

File: tests/features/content-row.feature [hunk 5]
```diff
@@ -16,6 +16,7 @@ Feature: Content row paragraph.
     And the "Paragraphs" field in the 1st "Content row" paragraph can reference:
       | Accordion             |
       | Contextual navigation |
+      | Social media follow   |
```
Comment (ademarco) [inline tests/features/content-row.feature]: This is ok since when we install the module we indeed get the paragraph there.

**Pull request merged**